packages/client/services/MpeService.ts

```diff
@@ -1,41 +1,54 @@
 import {
-    LibraryMpeRoomSearchResponseBody,
-    MtvRoomSearchRequestBody,
+    MpeSearchMyRoomsRequestBody,
+    MpeSearchMyRoomsResponseBody,
+    ListAllMpeRoomsResponseBody,
+    ListAllMpeRoomsRequestBody,
 } from '@musicroom/types';
 import urlcat from 'urlcat';
 import redaxios from 'redaxios';
 import { SERVER_ENDPOINT } from '../constants/Endpoints';
 
-export async function fetchLibraryMpeRooms({
-    userID,
-}: {
+interface FetchLibraryMpeRoomsArgs {
     userID: string;
-}): Promise<LibraryMpeRoomSearchResponseBody> {
-    const url = urlcat(SERVER_ENDPOINT, '/mpe/search/user-rooms');
+    searchQuery: string;
+    page: number;
+}
 
-    const rawResponse = await redaxios.post(url, {
+export async function fetchLibraryMpeRooms({
+    userID,
+    searchQuery,
+    page,
+}: FetchLibraryMpeRoomsArgs): Promise<MpeSearchMyRoomsResponseBody> {
+    const url = urlcat(SERVER_ENDPOINT, '/mpe/search/my-rooms');
+    const body: MpeSearchMyRoomsRequestBody = {
         userID,
-    });
-    const parsedResponse = LibraryMpeRoomSearchResponseBody.parse(
-        rawResponse.data,
-    );
+        searchQuery,
+        page,
+    };
+
+    const rawResponse = await redaxios.post(url, body);
+    const parsedResponse = MpeSearchMyRoomsResponseBody.parse(rawResponse.data);
```

Comment:
This is a real pleasure to red
So few lines for a lot of work

Reply:
Haha yeah! Love it too.